scripts/test.sh

```diff
@@ -508,7 +508,7 @@ function bom_pass {
 
 function dump_deps_of_module() {
   local module
-  if ! module=$(run go list -m); then
+  if ! module=$(run go mod edit -json | jq -r .Module.Path); then
```

Comment:
Could we merge it independent to migration to workspaces as a separate PR? Having less changes in the main PR will increase confidence during review.

Reply:
Great idea. Thanks, Marek :bow:. I'll break down this pull request even further, as most of the commits can coexist with the current codebase.

Reply:
I opened #19463.